test/unit/math/rev/prob/neg_binomial_2_test.cpp

```diff
@@ -1,20 +1,500 @@
 #include <stan/math/rev/scal.hpp>
+#include <boost/math/tools/numerical_differentiation.hpp>
+#include <boost/math/special_functions/digamma.hpp>
 #include <gtest/gtest.h>
 #include <vector>
+#include <algorithm>
 
-TEST(ProbDistributionsNegBinomial, derivatives) {
+namespace neg_binomial_2_test_internal {
+struct TestValue {
+  unsigned int n;
+  double mu;
+  double phi;
+  double value;
+  double grad_mu;
+  double grad_phi;
+
+  TestValue(unsigned int _n, double _mu, double _phi, double _value,
+            double _grad_mu, double _grad_phi)
+      : n(_n),
+        mu(_mu),
+        phi(_phi),
+        value(_value),
+        grad_mu(_grad_mu),
+        grad_phi(_grad_phi) {}
+};
+
+// Test data generated in Mathematica (Wolfram Cloud). The code can be re-ran
+// at https://www.wolframcloud.com/env/martin.modrak/NegBinomial2_Tests.nb
+// but is also presented below for convenience:
+//
+// nb2[n_,mu_,phi_]:= LogGamma[n + phi] - LogGamma[n + 1] - LogGamma[phi ]+
+//   n * (Log[mu] - Log[mu + phi]) + phi * (Log[phi] - Log[mu + phi])
+// nb2dmu[n_,mu_,phi_]= D[nb2[n, mu, phi],mu];
+// nb2dphi[n_,mu_,phi_]= D[nb2[n, mu, phi],phi];
+// out = OpenWrite["nb_test.txt"]
+// mus= {256*10^-7,314*10^-3,15*10^-1,8,180,  1123,10586};
+// phis=  {4*10^-4,65*10^-3,442*10^-2,800, 15324,150000};
+// ns = {0,6,14,1525,10233};
+//  WriteString[out, "std::array<TestValue, ",
+//      Length[mus]*Length[phis]*Length[ns], "> testValues = {"];
+//    Block[{$MaxPrecision = 80, $MinPrecision = 40}, {
+//      For[i = 1, i <= Length[mus], i++, {
+//        For[j = 1, j <= Length[phis], j++, {
+//        For[k = 1, k <= Length[ns], k++, {
+//          cmu = mus[[i]];
+//          cphi = phis[[j]];
+//     cn=ns[[k]];
+//          val = N[nb2[cn,cmu,cphi]];
+//     ddmu= N[nb2dmu[cn,cmu,cphi]];
+//     ddphi= N[nb2dphi[cn,cmu,cphi]];
+//          WriteString[out,"  TestValue(",CForm[cn],",",CForm[cmu],",",
+//            CForm[cphi],",", CForm[val],","CForm[ddmu],",",CForm[ddphi],"),"]
+//        }]
+//      }]
+//   }]
+//    }];
+//  WriteString[out,"};"];
+//  Close[out];
+//  FilePrint[%]
+std::array<TestValue, 210> testValues = {
```

Comment:
When the size gets to 210, this should be `std::vector`---it's way too many to count by hand.

Reply:
Modified.